packages/monaco/src/browser/monaco-text-model-service.ts

```diff
@@ -14,13 +14,27 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { inject, injectable } from 'inversify';
+import { inject, injectable, named } from 'inversify';
 import { MonacoToProtocolConverter, ProtocolToMonacoConverter } from 'monaco-languageclient';
 import URI from '@theia/core/lib/common/uri';
-import { ResourceProvider, ReferenceCollection, Event } from '@theia/core';
+import { ResourceProvider, ReferenceCollection, Event, MaybePromise, Resource, ContributionProvider } from '@theia/core';
 import { EditorPreferences, EditorPreferenceChange } from '@theia/editor/lib/browser';
 import { MonacoEditorModel } from './monaco-editor-model';
 
+export const MonacoEditorModelFactory = Symbol('MonacoEditorModelFactory');
+export interface MonacoEditorModelFactory {
+
+    readonly scheme: string;
+
+    createModel(
+        resource: Resource,
+        m2p: MonacoToProtocolConverter,
```

Comment:
I don't think we need to pass converters. They can be injected by subclasses. Or we can convert interface to abstract class to replace interface + symbol and inject converters for all subclasses.